### DISCUSSION TITLE
File upload from NextJS API route using multiparty

### DISCUSSION THREAD

**Discussion by Lyn-2004** [vercel/next.js#52635] [Help] [answered]
### Summary

I need help with my diploma project, I want to upload pictures of my products on to the website, but it is not working and I don't know why. 
This is where I'm sending the api request
`import axios from "axios";
import { useRouter } from "next/router";
import { useState } from "react";
import FormData from 'form-data';

export default function ProductForm({
    _id,
    titel : existingTitel,
    description : existingDescription,
    price : existingPrice,
    images,
}){
   
    const [titel, setTitel] = useState(existingTitel || '');
    const [description, setDescription] = useState(existingDescription || '');
    const [price, setPrice] = useState(existingPrice || '');
    const [goToProducts, setGoToProducts] = useState(false);
    const router = useRouter();

    async function saveProduct(ev){
        ev.preventDefault();
        const data = {titel,description, price}
        if(_id)
        {//update
            await axios.put('/api/products',{...data,_id});
            
        } else
        {//create
        await axios.post('/api/products', data)
        }
        setGoToProducts(true);
    }

    if (goToProducts)
    {
        router.push('/products')
    }
    async function uploadImages(ev){
        const files = ev.target?.files;
        if(files?.length > 0)
        {
            const data = new FormData();
            for (const file of files)
            {
                data.append('file',file);
            }
           const res = await fetch('/api/upload', {
            method: 'POST',
            body: data
        });
         console.log(res);
        }
    }

    return(
            <form onSubmit={saveProduct}>
            <label>Product name</label>
            <input 
            type="text"
            placeholder="product name" 
            value={titel} 
            onChange={ev => setTitel(ev.target.value)}
            />

            <label >
                Photos
            </label>

            <div className="mb-2">
                <label className="w-24 h-24 text-center flex items-center justify-center text-sm gap-1 text-gray-500 rounded-lg bg-gray-200">
                    <svg xmlns="http://www.w3.org/2000/svg" fill="none" viewBox="0 0 24 24" strokeWidth="1.5" stroke="currentColor" className="w-6 h-6">
                    <path strokeLinecap="round" strokeLinejoin="round" d="M3 16.5v2.25A2.25 2.25 0 005.25 21h13.5A2.25 2.25 0 0021 18.75V16.5m-13.5-9L12 3m0 0l4.5 4.5M12 3v13.5" />
                    </svg>
                    Upload
                    <input
                    type="file" onChange={uploadImages} className="hidden"
                    ></input>
                </label>

                {!images?.length &&
                (
                <div>
                    No photos in this products
                </div>)}

            </div>

            <label>Description</label>
            <textarea 
            placeholder="description"
            value={description}
            onChange={ev => setDescription(ev.target.value)}
            /> 
            <label>Price (in Kenya Schilling)</label>
            <input
            type="number" 
            placeholder="price"
            value={price}
            onChange={ev => setPrice(ev.target.value)}
            />
            <button 
            type="submit" className="btn-primary">Save</button>
            </form>

    );
}`
And this is the uploud.js 
`import multiparty from 'multiparty';
import {PutObjectCommand, S3Client} from '@aws-sdk/client-s3';
import fs from 'fs';
import mime from 'mime-types';
const bucketName ='eroots-next-ecommerce';
export default async function handel(req,res){

   const form = multiparty.Form();
    const {fields,files}= await new Promise((resolve,reject) => { 
        form.parse(req,(err,fields,files)=>{
        if (err) reject(err);
        resolve({fields,files});
       });
    });
   
    const client = new S3Client({
        region:'eu-north-1',
        credentials:{
            accessKeyId:process.env.S3_ACCESS_KEY,
            secretAccessKey:process.env.S3_SECRET_ACCESS_KEY,   
        },
    });

    const links=[];
    for(const file of files.file)
    { 
        const ext = file.originalFilename.split('.').pop();
        const newFilename = Date.now() +'.'+ext;
        client.send(new PutObjectCommand(
        {
            Bucket: bucketName,
            Key: newFilename,
            Body: fs.readFileSync(file.path),
            ACL:'public-read',
            ContentType: mime.lookup(file.path),
        }
        ));

        const link = `https://${bucketName}.s3.amazonaws.com/${newFilename}`;
        links.push(link);
    }
    
    console.log('length:', files);
    console.log(fields);

    return res.json({links});
}

export const config = {
    api: {bodyParser:false},
};`
Please I need help

**Comment by icyJoseph** [comment]
That's a lot of code and not very well formatted, but could you do a few checks:

1. In the network tab, make sure that the post request contains the binary data associated with the files
2. In the server, if the browser is sending the data, are the other fields and files arriving?

Or in other words, what is not working exactly?

**Reply by Lyn-2004** [reply, marked as answer]
I've found the mistake it was because I was not creating a new multiparty
